batch/batch/worker/worker.py

```diff
@@ -124,15 +125,23 @@ def deeper_stack_level_warn(*args, **kwargs):
 log.info(f'INSTANCE_CONFIG {INSTANCE_CONFIG}')
 log.info(f'INSTANCE_ENVIRONMENT {INSTANCE_ENVIRONMENT}')
 log.info(f'MAX_IDLE_TIME_MSECS {MAX_IDLE_TIME_MSECS}')
+log.info(f'INTERNET_INTERFACE {INTERNET_INTERFACE}')
 log.info(f'UNRESERVED_WORKER_DATA_DISK_SIZE_GB {UNRESERVED_WORKER_DATA_DISK_SIZE_GB}')
 
 instance_config = instance_config_from_config_dict(INSTANCE_CONFIG)
 assert instance_config.cores == CORES
 assert instance_config.cloud == CLOUD
 
-deploy_config = DeployConfig('gce', NAMESPACE, {})
+if CLOUD == 'gcp':
+    deploy_location = 'gce'
```

Comment:
This feels like a rake waiting to slam us in the face.

Reply:
Yeah, sorry, I missed this earlier. There are only three "deploy locations":

1. Inside a pod inside a Kubernetes cluster
2. Inside a Batch job
3. Everywhere else

The right fix here is to use better names, such as: `k8s`, `batch-job`, and `external`. The short term fix is to just pervasively use the word `gce` and eliminate `azure` as an option for DeployConfig.

Reply:
The reasoning for this is because there are three (classes of) names for any service inside k8s:

1. service.namespace: this is only accessible inside k8s
2. service.hail: this is a special DNS name used when you're in the cloud but outside k8s (the internal gateway is cheaper & does not expose traffic to the internet, so it's important to distinguish (2) from (3))
3. service.hail.is: this is accessible from the Internet